glue/core/roi.py

```diff
@@ -561,6 +588,70 @@ def move_to(self, xdelta, ydelta):
         self.vy = list(map(lambda y: y + ydelta, self.vy))
 
 
+def _project(projection_matrix, x, y, z):
+    """Projects 3d coordinates to 2d coordinates using a 4x4 matrix"""
+    if not isinstance(x, np.ndarray):
+        x = np.asarray(x)
```

Comment:
I think you can get rid of `if not isinstance(x, np.ndarray):` as I think `np.asarray` doesn't do a copy if the input is already an array?